dispatch/static/manager/src/js/vendor/dispatch-editor/embeds/TwitterEmbed.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+
+import fetchJsonp from 'fetch-jsonp'
+import url from 'url'
+
+import { Button } from '@blueprintjs/core'
+import { FormInput, TextInput } from '../../../components/inputs'
+
+require('isomorphic-fetch')
+require('../styles/embeds/video.scss')
+
+const TWITTER_API_URL = 'https://api.twitter.com/1.1/statuses/oembed.json'
+
+class TwitterEmbedComponent extends React.Component {
+
+  validateTweetURL(url) {
+    return /^(?:https?:\/\/)?(?:www\.)?twitter\.com\/?(\w+)\/(status)\/(\d+)/g.test(url)
+  }
+
+  getTweet() {
+    const that = this
+    if (this.validateTweetURL(this.props.data.url)) {
+      fetchJsonp(TWITTER_API_URL + url.format({query: {url: this.props.data.url}}))
+      .then((response) => response.json())
+      .then(data => {
+        that.props.updateField('tweet', data.html)
+        this.props.stopEditing()
```

Comment:
I'm getting this error when I click "Insert":

![screen shot 2018-01-19 at 4 36 06 pm](https://user-images.githubusercontent.com/2547035/35177830-f92a49d8-fd37-11e7-9776-b18eb1013460.png)

I suspect that it can be solved by calling `that.props.stopEditing()` instead of using `this`

Reply:
anyone else think its weird the fetch isn't in a thunk? just me?

Reply:
@JimFung would you mind expanding on that?

Reply:
@lorenzodb1 Peter and I talked about it when I visited a few days ago. I wouldn't worry about it, as it is now matches the design pattern that Peter wants for the embeds 👍